org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/tracking/PersistentIssueTrackerCache.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * SonarLint for Eclipse
+ * Copyright (C) 2015-2016 SonarSource SA
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.core.internal.tracking;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class PersistentIssueTrackerCache implements IssueTrackerCache {
```

Comment:
Not sure how this can be better. `IssueTrackerCache` is the interface needed by the `IssueTracker`. This class does just that, using a combination of an in-memory and a persistent store.